### PR TITLE
stillAlive option changes and documentation

### PR DESCRIPTION
**Pull request by chrisdepas**
## stillAlive changes & documentation
- Added stillAlive info to docs
- Changed server example to refer to `open-easyrtc` rather than `easyrtc`
- Separated `stillAliveInterval` and `stillAliveGracePeriod` options
    - Setting `stillAliveInterval` to 0 (or null) will now disable timeouts, matching comment in the default options file
    - Previously, both `stillAliveInterval` and `stillAliveGracePeriod` would need to be set to 0 to disable timeouts (`stillAliveInterval` disabled sending of stillAlive packets by clients, `stillAliveGracePeriod` disabled tracking of timeouts on the signalling server)

I'm not sure that I understand what the intent was for these options before, my best guess is that it was set up so that the interval reported to clients and the interval used for tracking timeouts could be changed independently.

My reason for this PR is that I have clients connecting from mobile devices, so I don't get the stillAlive functionality for free. When I tried to disable it, I found that I would have to set both of the stillAlive options to zero.

## Custom Error Handler
- Added `setErrorListener` function, which can be used to override the default Open-EasyRTC error handling
    - This allows us to avoid displaying the EasyRTC error popup completely, and e.g. a materialize toast can be used instead
    - Also added the changes to the typescript definition file

## Documentation & Cleanup
- Removed more bower references
- Added info to `api/README.md` on unit tests & linting

### CODE REVIEW

**Comment by hthetiot** [comment]
@chrisdepas Thank you for your PR. 
Let me know when you want me to review and merge.

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ Documentation
 -------------
 All documentation can be found within [the docs folder](./docs/).
 
+_**Note:** Links marked with an asterisk (__*__) use the third-party service [http://htmlpreview.github.io](http://htmlpreview.github.io) to render HTML._
```
Comment (chrisdepas) [inline README.md]: Not sure if this is a good idea, please let me know if you are against it and I can remove these

File: docs/docs_in_html/easyrtc_faq.html [hunk 2]
```diff
@@ -95,7 +95,7 @@ <h2>EasyRTC Requirements:</h2>
 
 <h2>Is EasyRTC production ready?</h2>
 
-<p>We have used EasyRTC in production since May of 2013 (see <a href="https://tawk.com" title="tawk.com">https://tawk.com</a>). The included demos include notes about what browsers the various features work in, or how experimental the feature is.</p>
```
Comment (chrisdepas) [inline docs/docs_in_html/easyrtc_faq.html]: This was presumably owned by Priologic, the link appears to be broken

File: docs/easyrtc_common_problems.md [hunk 3]
```diff
@@ -71,74 +62,4 @@ The demos at demo.EasyRTC.com are backed by a turn server and should still work
...
-that happen in a single thread and lots of people have used it directly. 
-The server code tends to be harder to follow and relatively few people have 
-bothered to learn how it works. 
-If you are changing/extending the server code, 
-there are going to be fewer people who can help you; 
-you may need to familarize youself with the server code yourself to be your own solution.
-
-If you are looking for a WebRTC/EasyRTC expert and no-one steps up, 
-that's possibly because they are busy with their own projects. 
+
```
Comment (chrisdepas) [inline docs/easyrtc_common_problems.md]: My reasons for removing this
- This seems like common sense/manners
- Anyone who would act like this (i.e. do the things it warns against) isn't likely to read the documentation
- Is a bit condescending for new people reading the documentation

If you think this is effective however, I can revert this

File: docs/easyrtc_faq.md [hunk 4]
```diff
@@ -42,36 +42,36 @@ For the most part, if you can run Node.JS, you can run EasyRTC.
...
+In the browser, you've got a single thread and no memory management issues.
+With native clients, one careless memory access from any other part of your program and you've got weird intermittent bugs
+that are time-consuming to track down.
+
+So we suggest using Cordova, which basically lets you write web applications that run inside a native app (both android and iOS).
+Actually, they are hybrid apps, you can mix in android calls as well.
+
+If you want to write your own EasyRTC native library and can think of a way to make it a viable business, feel free to.
+
+
```
Comment (chrisdepas) [inline docs/easyrtc_faq.md]: I don't think I changed this, this is probably git replacing newline characters

File: docs/webrtc_debugging.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+```
+More information on logging in Chrome in general can be found [here](https://www.chromium.org/for-testers/enable-logging), and WebRTC-specific logging information can be found [here](https://webrtc.org/web-apis/chrome/#native-webrtc-logging-in-chrome).
+
+### Firefox
+
+Firefox has a built-in debugging tool, which can be found by entering __*about:webrtc*__ into your url bar.
+
+Additionally, a plugin can be installed to provide extra features, more info can be found in [the Mozilla blog](https://blog.mozilla.org/webrtc/new-tool-debugging-webrtc/).
+
+Logging can be enabled via the __*about:networking#logging*__ config page. Logging must be enabled specifically for each module you wish to debug. Information on which modules you need to enable can be found on [the Mozilla wiki](https://wiki.mozilla.org/Media/WebRTC/Logging).
```
Comment (chrisdepas) [inline docs/webrtc_debugging.md]: This is completely new, I omitted the 'easyrtc_' prefix on the filename to distinguish it from Priologic's existing documentation

**Comment by chrisdepas** [comment]
@hthetiot Thanks, this is ready for review when you have time. 

It passes linting, but I haven't been able to get the tests working (seems like some sort of concurrency issue)

File: README.md (hunk 1 above)
Comment (hthetiot) [reply README.md]: I like that.

File: docs/docs_in_html/easyrtc_faq.html (hunk 2 above)
Comment (hthetiot) [reply docs/docs_in_html/easyrtc_faq.html]: LGTM

File: docs/easyrtc_common_problems.md (hunk 3 above)
Comment (hthetiot) [reply docs/easyrtc_common_problems.md]: LGTM

**Comment by hthetiot** [comment]
I need more time to review, will 'eep you posted.
Thank you @chrisdepas

**Comment by chrisdepas** [comment]
@hthetiot I've merged in other branches branches as you requested, this is ready for review when you have time
I've tried to merge in changes since my initial fork, hopefully I've done this correctly and it doesn't cause any headaches :sweat_smile:

**Comment by hthetiot** [comment]
@chrisdepas Let merge that in beta and I will make some testing later.

**Comment by hthetiot** [comment]
@chrisdepas I also sent you an invite to collaborate on this repository.
I will release 2.0.3 with #16 and #15 and we can release your changes in 2.1.0 if you are ok with that.

**Comment by chrisdepas** [comment]
@hthetiot Thanks for the collaboration invite. I won't have time to do anything for at least the next two weeks, but I'm happy to contribute after I return to my normal work tasks.

I also noticed that you merged these into master, thanks!

**Comment by hthetiot** [comment]
@chrisdepas Sure do when you have time, It's not paid anyway :)
It's open-source you can do it when you want and if you want no string attached.

**Pull request merged**